packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart

```diff
@@ -425,16 +427,42 @@ class GtagExceptionDevTools extends GtagException {
   external int? get inspector_tree_controller_id;
 }
 
-/// Request DevTools property value 'enabled' (GA enabled) stored in the file
-/// '~/.flutter-devtools/.devtools'.
+/// Whether google analytics are enabled.
 Future<bool> isAnalyticsEnabled() async {
-  return await server.isAnalyticsEnabled();
+  bool enabled = false;
+  if (kReleaseMode) {
+    enabled = await dtdManager.analyticsTelemetryEnabled();
+  }
+
+  // TODO(https://github.com/flutter/devtools/issues/7083): remove this block
+  // when the legacy analytics are fully removed. For now, check that both
+  // unified analytics are enabled and the legacy analytics are enabled.
+  if (enabled) {
+    enabled = await server.isAnalyticsEnabled();
+  }
+
+  return enabled;
 }
 
-/// Set the DevTools property 'enabled' (GA enabled) stored in the file
-/// '~/flutter-devtools/.devtools'.
-Future<bool> setAnalyticsEnabled(bool value) async {
-  return await server.setAnalyticsEnabled(value);
+/// Whether the google analytics consent message should be shown.
+Future<bool> shouldShowAnalyticsConsentMessage() async {
+  bool shouldShow = false;
+  if (kReleaseMode) {
+    // When asked if the consent message should be shown,
+    // package:unified_analytics will return true if this the user's first run
+    // (unlikely for the case of DevTools since the user is likely to have
+    // already seen the message from another Dash tool (Dart or Flutter CLI,
+    // IDE, etc.) or when the consent message version has been updated.
+    shouldShow = await dtdManager.shouldShowAnalyticsConsentMessage();
+  }
```

Comment:
This comment may be wrong, even if a user has already interacted with package:unified_analytics through the flutter-cli, we will still have to show the consent message because it is the first time it was run on devtools.

So the package will return true if this is the first time we added devtools into the config file for UA

Reply:
Updated comment.